GLCI [test-win-dev][test-win-rel]: Fail to install with 'gcc not found'

Links to jobs:

https://gitlab.com/Rdatatable/data.table/-/jobs/7949158879
https://gitlab.com/Rdatatable/data.table/-/jobs/7949158861

Links to 00install.out:

https://gitlab.com/Rdatatable/data.table/-/jobs/7949158879/artifacts/file/bus/test-win-dev/data.table.Rcheck/00install.out
https://gitlab.com/Rdatatable/data.table/-/jobs/7949158861/artifacts/file/bus/test-win-rel/data.table.Rcheck/00install.out

One 00install.out:

```
* installing *source* package 'data.table' ...
** using staged installation

   **********************************************
   WARNING: this package has a configure script
         It probably needs manual configuration
   **********************************************


** libs
gcc  -I"C:/R/include" -DNDEBUG     -I"c:/rtools44/x86_64-w64-mingw32.static.posix/include"   -fopenmp   -O2 -Wall  -mfpmath=sse -msse2 -mstackrealign  -c assign.c -o assign.o
/bin/sh: line 1: gcc: command not found
make: *** [C:/R/etc/x64/Makeconf:290: assign.o] Error 127
ERROR: compilation failed for package 'data.table'
* removing 'C:/GitLab-Runner/builds/Rdatatable/data.table/data.table.Rcheck/data.table'
```

--- a/.gitlab-ci.yml
+++ b/.gitlab-ci.yml
@@ -16,14 +16,14 @@ variables:
   R_REL_VERSION:  "4.4"
   R_REL_WIN_BIN:  "https://cloud.r-project.org/bin/windows/base/old/4.4.1/R-4.4.1-win.exe"
   RTOOLS_REL_BIN: "https://cloud.r-project.org/bin/windows/Rtools/rtools44/files/rtools44-6104-6039.exe"
-  RTOOLS44_HOME:  "/c/rtools"
+  RTOOLS44_HOME:  "/c/rtools" # needed for finding gcc #6552
   R_DEV_VERSION:  "4.5"
   R_DEV_WIN_BIN:  "https://cloud.r-project.org/bin/windows/base/R-devel-win.exe"
-  RTOOLS_DEV_BIN: "https://cloud.r-project.org/bin/windows/Rtools/rtools44/files/rtools44-6104-6039.exe"
+  RTOOLS_DEV_BIN: "https://cloud.r-project.org/bin/windows/Rtools/rtools44/files/rtools44-6104-6039.exe" # dev uses same rtools as release
   R_OLD_VERSION:  "4.3"
   R_OLD_WIN_BIN:  "https://cloud.r-project.org/bin/windows/base/old/4.3.3/R-4.3.3-win.exe"
   RTOOLS_OLD_BIN: "https://cloud.r-project.org/bin/windows/Rtools/rtools43/files/rtools43-5958-5975.exe"
-  RTOOLS43_HOME:  "/c/rtools"
+  RTOOLS43_HOME:  "/c/rtools" # needed for finding gcc #6552
 
 stages:
   - dependencies
@@ -227,8 +227,6 @@ test-lin-ancient-cran:
     - Rscript.exe -e "source('.ci/ci.R'); install.packages(dcf.dependencies('DESCRIPTION', which='all'), repos=file.path('file://',getwd(),'bus/mirror-packages/cran'), quiet=TRUE)"
     - cp.exe $(ls.exe -1t bus/build/data.table_*.tar.gz | head.exe -n 1) .
   script:
-    - echo $env:PATH
-    - Rscript.exe -e "Sys.which('gcc')"
     - R.exe CMD check --no-manual $(ls.exe -1t data.table_*.tar.gz | head.exe -n 1)
     - R.exe CMD INSTALL --build $(ls.exe -1t data.table_*.tar.gz | head.exe -n 1)
     - -not (grep.exe "warning:" data.table.Rcheck\00install.out)

--- a/.gitlab-ci.yml
+++ b/.gitlab-ci.yml
@@ -16,15 +16,14 @@ variables:
   R_REL_VERSION:  "4.4"
   R_REL_WIN_BIN:  "https://cloud.r-project.org/bin/windows/base/old/4.4.1/R-4.4.1-win.exe"
   RTOOLS_REL_BIN: "https://cloud.r-project.org/bin/windows/Rtools/rtools44/files/rtools44-6104-6039.exe"
-  RTOOLS43_HOME:  "/c/rtools"
+  RTOOLS44_HOME:  "/c/rtools"
   R_DEV_VERSION:  "4.5"
   R_DEV_WIN_BIN:  "https://cloud.r-project.org/bin/windows/base/R-devel-win.exe"
   RTOOLS_DEV_BIN: "https://cloud.r-project.org/bin/windows/Rtools/rtools44/files/rtools44-6104-6039.exe"
-  RTOOLS44_HOME:  "" ## in case R-devel will use new Rtools toolchain, now it uses 4.4 env var
   R_OLD_VERSION:  "4.3"
   R_OLD_WIN_BIN:  "https://cloud.r-project.org/bin/windows/base/old/4.3.3/R-4.3.3-win.exe"
   RTOOLS_OLD_BIN: "https://cloud.r-project.org/bin/windows/Rtools/rtools43/files/rtools43-5958-5975.exe"
-  RTOOLS42_HOME:  "/c/rtools"
+  RTOOLS43_HOME:  "/c/rtools"
 
 stages:
   - dependencies
@@ -228,6 +227,8 @@ test-lin-ancient-cran:
     - Rscript.exe -e "source('.ci/ci.R'); install.packages(dcf.dependencies('DESCRIPTION', which='all'), repos=file.path('file://',getwd(),'bus/mirror-packages/cran'), quiet=TRUE)"
     - cp.exe $(ls.exe -1t bus/build/data.table_*.tar.gz | head.exe -n 1) .
   script:
+    - echo $env:PATH
+    - Rscript.exe -e "Sys.which('gcc')"
     - R.exe CMD check --no-manual $(ls.exe -1t data.table_*.tar.gz | head.exe -n 1)
     - R.exe CMD INSTALL --build $(ls.exe -1t data.table_*.tar.gz | head.exe -n 1)
     - -not (grep.exe "warning:" data.table.Rcheck\00install.out)

--- a/CODEOWNERS
+++ b/CODEOWNERS
@@ -68,3 +68,6 @@
 # docs
 /man/openmp-utils.Rd @Anirban166
 /Seal_of_Approval.md @tdhock
+
+# GLCI
+.gitlab-ci.yml @jangorecki @ben-schwen

